src/main/groovy/io/openliberty/tools/gradle/tasks/DevTask.groovy

```diff
@@ -681,6 +741,29 @@ class DevTask extends AbstractServerTask {
         if (pollingTest == null) {
             pollingTest = DEFAULT_POLLING_TEST;
         }
+
+        if (container == null) {
+            boolean buildContainerSetting = project.liberty.dev.container; // get from build.gradle or from -Pdev_mode_container=true
+            if (buildContainerSetting == null) {
+                setContainer(DEFAULT_CONTAINER);
+            } else {
+                setContainer(buildContainerSetting);
+            }
+        }
+
+        if (dockerfile == null) {
+            File buildDockerfileSetting = project.liberty.dev.dockerfile; // get from build.gradle
+            if (buildDockerfileSetting != null) {
+                setDockerfile(buildDockerfileSetting.getAbsolutePath()); // setDockerfile will convert it to canonical path
+            }
+        }
+
+        if (dockerRunOpts == null) {
+            String buildDockerRunOptsSetting = project.liberty.dev.dockerRunOpts; // get from build.gradle
+            if (buildDockerRunOptsSetting != null) {
+                setDockerRunOpts(buildDockerRunOptsSetting);
```

Comment:
Should this set container to true? I'm thinking the setDockerfile and setDockerRunOpts methods should just call setContainer(true). Would that make sense?

Reply:
This only handles when the dockerRunOpts or dockerfile params are not set in command line but set in the dev extension.  We also need to address when the params are set in the command line.  The logic in processContainerParams handles both cases.